rust/lance/src/index/vector/ivf/builder.rs

```diff
@@ -92,6 +92,7 @@ pub async fn shuffle_dataset(
             .await
             .map_err(|e| Error::Index {
                 message: e.to_string(),
+                location: location!(),
```

Comment:
Can we reuse the inner error's location here?

Reply:
Here `e` is from `tokio::task::JoinError`, it lasks of location.